### PR TITLE
Turn exercise 006 into practice #2172

### PR DESCRIPTION
**Pull request by Appleeyes**
This PR is referencing the issue: Turn exercise 006 into practice #2172

Please help me review and please incase I miss out anything let me know @SaySayo @cuihtlauac

### CODE REVIEW

File: opam-tailwindcss.tar.gz
Comment (cuihtlauac) [inline opam-tailwindcss.tar.gz:1]: Remove this file from the PR

File: practice/006/answer/impl.ml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+let is_palindrome list =
+    list = List.rev list
+
```
Comment (cuihtlauac) [inline practice/006/answer/impl.ml]: Add newline at end of file.

File: practice/006/work/impl.ml [hunk 2]
```diff
@@ -0,0 +1,3 @@
+let is_palindrome list =
+  list = List.rev list
+
```
Comment (cuihtlauac) [inline practice/006/work/impl.ml]: Add newline at end of file

File: practice/006/ex.ml [hunk 3]
```diff
@@ -0,0 +1,17 @@
+
```
Comment (cuihtlauac) [inline practice/006/ex.ml]:
```suggestion
```

File: practice/006/work/impl.ml [hunk 4]
```diff
@@ -0,0 +1,3 @@
+let is_palindrome list =
+  list = List.rev list
```
Comment (cuihtlauac) [inline practice/006/work/impl.ml]: This should throw an exception.

**Review by cuihtlauac — CHANGES_REQUESTED** [review]
Thanks @Appleeyes, this is nice.
- File `work/test/run.ml` is missing
- File `work/impl.ml` does not have the correct content
- File `opam-tailwindcss.tar.gz` does not belong to the PR

File: opam-tailwindcss.tar.gz
Comment (Appleeyes) [reply opam-tailwindcss.tar.gz:1]: Done

File: practice/006/answer/impl.ml (hunk 1 above)
Comment (Appleeyes) [reply practice/006/answer/impl.ml]: Done

File: practice/006/ex.ml (hunk 3 above)
Comment (Appleeyes) [reply practice/006/ex.ml]: Done

File: practice/006/work/impl.ml (hunk 4 above)
Comment (Appleeyes) [reply practice/006/work/impl.ml]: Done

**Comment by Appleeyes** [comment]
> Thanks @Appleeyes, this is nice.
> 
> * File `work/test/run.ml` is missing
> * File `work/impl.ml` does not have the correct content
> * File `opam-tailwindcss.tar.gz` does not belong to the PR

Everything has been modified

File: practice/006/answer/impl.ml (hunk 1 above)
Comment (cuihtlauac) [inline practice/006/answer/impl.ml]:
```suggestion

```

File: practice/006/ex.ml [hunk 5]
```diff
@@ -0,0 +1,16 @@
+open OUnit2
+
+module type Testable = sig
+  val is_palindrome : 'a list -> bool
+end
+
+module Make(Tested: Testable) : sig val v : test end = struct
+  let v = "is_palindrome" >::: [
+    "palindrome" >:: (fun _ -> assert_equal true (Tested.is_palindrome ["x"; "a"; "m"; "a"; "x"]));
```
Comment (cuihtlauac) [inline practice/006/ex.ml:9]: Add a test for the empty list

**Review by cuihtlauac — CHANGES_REQUESTED** [review]

File: practice/006/answer/impl.ml [hunk 6]
```diff
@@ -0,0 +1,2 @@
+let is_palindrome list =
+    list = List.rev list
```
Comment (cuihtlauac) [inline practice/006/answer/impl.ml]:
```suggestion
    list = List.rev list

```

**Review by cuihtlauac — CHANGES_REQUESTED** [review]

**Review by cuihtlauac — APPROVED** [review]
Thanks @Appleeyes
